extensions/parquet/base/src/main/java/io/deephaven/parquet/base/util/SeekableChannelsProvider.java

```diff
@@ -3,20 +3,51 @@
  */
 package io.deephaven.parquet.base.util;
 
+import io.deephaven.UncheckedDeephavenException;
+import io.deephaven.util.SafeCloseable;
+import io.deephaven.util.annotations.FinalDefault;
 import org.jetbrains.annotations.NotNull;
 
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.channels.SeekableByteChannel;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
 public interface SeekableChannelsProvider {
 
-    default SeekableByteChannel getReadChannel(@NotNull final String path) throws IOException {
-        return getReadChannel(Paths.get(path));
```

Comment:
This is a breaking change; I'm not sure we need to remove it?

Reply:
Agreed, this is a breaking change. But we confirmed on the group this class is only used for parquet stuff. 
And we need to pass the context object, so I have added another API which takes context and string both.